### PR TITLE
Simplify Fish handler

### PR DESCRIPTION
**Pull request by kidonng**
The handler function for fish is (mostly) written almost six years ago and looks unnecessarily complex today. This PR aims to make it modern and simpler.

Details about the changes will be later added as inline comments.

### CODE REVIEW

File: handler.fish [hunk 1]
```diff
@@ -1,24 +1,14 @@
-function __fish_command_not_found_on_interactive --on-event fish_prompt
```
Comment (kidonng) [inline handler.fish:1]: Searching around GitHub and Google suggests me that this is only used by Homebrew. I don't know its original purpose, but looks unnecessary now.

File: handler.fish [hunk 2]
```diff
@@ -1,24 +1,14 @@
...
-    functions --erase __fish_command_not_found_handler
```
Comment (kidonng) [inline handler.fish:2]: There is no need to erase old functions, the new one will replace the old one.

File: handler.fish [hunk 3]
```diff
@@ -1,24 +1,14 @@
...
-    functions --erase __fish_command_not_found_setup
```
Comment (kidonng) [inline handler.fish:3]: This is [removed](https://github.com/fish-shell/fish-shell/commit/1a12071a48a4ead52c8749118aef2a1016bdc5e5#diff-0d2c4442bc9780ab3f6d9e4b371b7cd7c89c18aca56bb25204e94d244a8d683b) prior-to Fish 3.0 (more than 5 years ago), so definitely not needed anymore.

File: handler.fish [hunk 4]
```diff
@@ -1,24 +1,14 @@
...
-    function __fish_command_not_found_handler --on-event fish_command_not_found
-        fish_command_not_found $argv
+function fish_command_not_found
+    set -l cmd $argv[1]
+    set -l txt (brew which-formula --explain $cmd 2> /dev/null)
+
+    if test -z "$txt"
+        __fish_default_command_not_found_handler $cmd
+    else
+        string collect $txt
```
Comment (kidonng) [inline handler.fish:8]: `string collect` is a relatively new thing added in [3.1b1](https://fishshell.com/docs/current/relnotes.html#fish-3-1b1-released-january-26-2020) which is close to 1.5 years ago. There could be people still using older versions but it should be rare.

**Comment by bfontaine** [comment]
Merged; thank you! :tada:

**Pull request merged**